src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+  data (): ChartPageData {
+    return {
+      chartData: {
+        labels: [],
+        datasets: [
+          {
+            label: '',
+            backgroundColor: [],
+            data: []
+          }
+        ]
+      },
+      categoryDataList: {},
+      pieChartData: {
+        labels: [],
+        datasets: [
+          {
+            backgroundColor: [],
+            data: []
+          }
+        ]
+      },
+      typeInformationPie: 'available',
+      typeInformationBar: ['allocated', 'spent', 'available'],
+      masterCategoryId: '',
+      currentGraph: 'pie',
+      budgetMonth: Time.getCurrentMonth()
+    }
+  },
+  computed: {
+    budget (): Budget | null {
+      return this.$store.state.budget
+    },
+    masterCategories (): GroupSelectOption[] {
+      const optionsList = [
+        {
+          label: this.$t('SELECT'),
+          options: [
+            { value: '', label: this.$t('ALL_BUDGET') }
+          ]
+        }
+      ]
+      for (const masterCategory of this.$store.state.masterCategories) {
+        const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, false)
+        if (categories.length > 0) {
+          const newOption = { value: masterCategory.id, label: masterCategory.name }
+          optionsList[0].options.push(newOption)
+        }
+      }
+      return optionsList
+    },
+    isThisYear (): boolean {
+      return Time.monthIsThisYear(this.budgetMonth)
+    }
+  },
+  methods: {
+    async getBudgetData () {
+      if (this.budget) {
+        await BudgetDataService.getBudgetDataForMonth(this.budget, this.budgetMonth).then(
+          (categoryDataList) => {
+            this.categoryDataList = categoryDataList
+          }
+        )
+      }
+    },
+    getNames (masterCategorySelectedId: string): string[] {
+      const listName: string[] = []
+      const masterCategorySelected = StoreHandler.getMasterCategoryById(this.$store, masterCategorySelectedId)
+      if (masterCategorySelected) {
+        const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategorySelected, false)
+        for (const category of categories) {
+          listName.push(category.name)
+        }
+      } else {
+        for (const masterCategory of this.$store.state.masterCategories) {
+          const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, false)
+          const archivedCategories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, true)
+          if (!(categories.length === 0 && archivedCategories.length > 0)) {
+            listName.push(masterCategory.name)
+          }
+        }
+      }
+      return listName
+    },
+    getDatas (type: string, masterCategorySelectedId: string): number[] {
+      const listData: number[] = []
+      const masterCategorySelected = StoreHandler.getMasterCategoryById(this.$store, masterCategorySelectedId)
+      if (masterCategorySelected) {
```

Comment:
this function would be more readable if divided in multiple smaller functions

Reply:
done